src/System.CommandLine/ParseResult.cs

```diff
@@ -48,7 +50,7 @@ public class ParseResult
 
         public IReadOnlyCollection<string> UnmatchedTokens { get; }
 
-        internal string RawInput { get; }
+        internal string RawInput => _rawInput ?? Environment.CommandLine;
```

Comment:
Is there a reason to do this check in the property's getter rather than just doing it once in the ctor? Is it possible for `Environment.CommandLine` to change during the life of the application?

Reply:
I made it lazy only because it might not be called. It's only used for suggestions.

Reply:
Gotcha. That makes sense.